extensions/email/src/main/java/org/mule/extension/email/internal/util/EmailConnectorUtils.java

```diff
@@ -0,0 +1,223 @@
+     *
+     * @param address the string to be converted.
+     * @return a new {@link InternetAddress} instance.
+     */
+    public static Address toAddress(String address)
+    {
+        try
+        {
+            return new InternetAddress(address);
+        }
+        catch (AddressException e)
+        {
+            throw new EmailException(format("Error while creating %s InternetAddress", address));
+        }
+    }
+
+    /**
+     * Converts a {@link List} of {@link String}s representing email addresses into an {@link InternetAddress} array.
+     *
+     * @param addresses the list to be converted.
+     * @return a new {@link Address}[] instance.
+     */
+    public static Address[] toAddressArray(List<String> addresses)
+    {
+        return addresses.stream().map(EmailConnectorUtils::toAddress).toArray(Address[]::new);
+    }
+
+    /**
+     * Extracts the incoming {@link MuleMessage} attributes of {@link EmailAttributes} type.
+     * <p>
+     * If no {@link EmailAttributes} are found in the incoming {@link MuleMessage}
+     * an exception with the {@code exceptionMessage} is thrown.
+     *
+     * @param muleMessage      the incoming {@link MuleMessage}.
+     * @param exceptionMessage i
+     * @return an value with the {@link EmailAttributes}.
+     */
+    public static EmailAttributes getAttributesFromMessage(MuleMessage muleMessage, String exceptionMessage)
+    {
+        if (muleMessage.getAttributes() instanceof EmailAttributes)
+        {
+            return (EmailAttributes) muleMessage.getAttributes();
+        }
+        throw new EmailException(format(ATTRIBUTES_NOT_FOUND_MASK, exceptionMessage));
+    }
+
+    /**
+     * Extracts the incoming {@link MuleMessage} attributes of {@link EmailAttributes} type.
+     * <p>
+     * If no {@link EmailAttributes} are found in the incoming {@link MuleMessage} an exception is thrown.
+     *
+     * @param muleMessage the incoming {@link MuleMessage}.
+     * @return an value with the {@link EmailAttributes}.
+     */
+    public static EmailAttributes getAttributesFromMessage(MuleMessage muleMessage)
+    {
+        return getAttributesFromMessage(muleMessage, "");
+    }
+
+    /**
+     * Extracts the text body of an email part.
+     *
+     * @param part the part to getPropertiesInstance the body from.
+     * @return the body of the part.
+     */
+    public static String getTextBody(Part part)
+    {
+        try
+        {
+            Object partContent = part.getContent();
+            if (isMultipart(partContent))
+            {
+                Multipart mp = (Multipart) partContent;
+                for (int i = 0; i < mp.getCount(); i++)
+                {
+                    BodyPart bodyPart = mp.getBodyPart(i);
+                    Object bodyPartContent = bodyPart.getContent();
+                    if (isText(bodyPartContent) && !isAttachment(bodyPart))
+                    {
+                        return bodyPartContent.toString().trim();
```

Comment:
why is trimming an important thing here? If it's really key, then set a variable and trim it once on the return so that you're sure that it always happens